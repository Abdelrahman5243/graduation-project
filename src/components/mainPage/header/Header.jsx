import React, { useState } from "react";
import "./header.css";
import logo from "../../../assets/images/mainPageAssets/logo.svg";
import hero from "../../../assets/images/mainPageAssets/hero.svg";
import posterImage from "../../../assets/images/mainPageAssets/signin.svg";
import BaseModal from "../modal/BaseModal/BaseModal.jsx";
import LoginForm from "../modal/LoginForm.jsx";
import SignUpForm from "../modal/SignUpForm.jsx";

const Header = () => {
  // for login/signin form
  const [isLogin, setIsLogin] = useState(false);

  const toggleForm = () => {
    setIsLogin(!isLogin);
  };

  const toggleModal = () => {
    document.querySelector(".modal-overlay").classList.toggle("closed");
  };
  return (
    <header className="main-header">
      <div className="container mx-auto px-4">
        <div className="head flex  justify-between items-center py-4">
          <div className="main-logo flex items-center">
            <img src={logo} alt="logo" />
            <span>WEB</span>
          </div>
          <button className="signIn-btn" onClick={toggleModal}>
            Sign In
          </button>
        </div>
        <div className="info-section flex gap-5 justify-between items-center py-8">
          <div className="hero-info">
            <div className="info-title">welcome to ZWEB where your websites dreams take flight with ai </div>
            <div className="info-description my-6">
              At Zzgazig Website Easy Builer, we're revolutionizing design proces With our intelligent AI website builder, your vision f or a digital presence
              is transformed into reality with just a few clicks. From nebulous ideas to fully-fledged websites, the future of web design begins here
            </div>
            <div className="info-btn">Generate your website</div>
          </div>
          <div className="main-hero-image">
            <img src={hero} alt="hero" />
          </div>
        </div>
      </div>

      <BaseModal poster={posterImage} toggleModal={toggleModal}>
        {isLogin ? <LoginForm toggleForm={toggleForm} /> : <SignUpForm toggleForm={toggleForm} />}
      </BaseModal>
    </header>
  );
};
export default Header;
