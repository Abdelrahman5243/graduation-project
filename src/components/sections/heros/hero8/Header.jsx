import React from "react";

const Header = ({ template }) => {
  const header = template.header;

  return (
    <div className="py-8">
      <div className="container mx-auto">
        <div className="flex justify-between items-center">
          {/* logo */}
          <a href="#">
            <img src={header.imgUrl} alt="picture of logo" />
          </a>
          <button className="btn btn-sm"> {header.buttonText} </button>
        </div>
      </div>
    </div>
  );
};

export default Header;
