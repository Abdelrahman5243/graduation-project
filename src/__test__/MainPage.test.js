/* eslint-disable no-unused-vars */
import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { useTranslation } from 'react-i18next';
import Header from '../components/mainPage/header/Header.jsx';

// Mocking the i18next hook
jest.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: (key) => key,
    i18n: {
      changeLanguage: jest.fn(),
    },
  }),
}));

// Define a mock initial state
const initialState = {
  auth: {
    user: {
      name: 'Test User',
      email: 'test@example.com',
    },
  },
  // Add other slices of state your components might use
};

// Create a mock reducer that returns the initial state
const mockReducer = (state = initialState, action) => state;

// Create the mock store with the initial state and reducer
const mockStore = createStore(mockReducer);

it('renders correctly', () => {
  const component = renderer.create(
    <Provider store={mockStore}>
      <Header toggleModal={() => {}} />
    </Provider>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
