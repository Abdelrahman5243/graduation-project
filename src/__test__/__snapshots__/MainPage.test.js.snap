// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<header
  className="main-header"
>
  <div
    className="container mx-auto px-4"
  >
    <div
      className="info-section flex gap-5 justify-between items-center py-8"
    >
      <div
        className="hero-info"
        data-aos="fade-right"
        data-aos-duration="1000"
      >
        <h1
          className="info-title"
        >
           
          HERO.TITLE
        </h1>
        <p
          className="info-description"
        >
           
          HERO.DESCRIPTION
           
        </p>
        <button
          className="info-btn"
          onClick={[Function]}
        >
          HERO.BUTTONTEXT
        </button>
      </div>
      <div
        className="main-hero-image"
      >
        <img
          alt="hero"
          src="hero-min.svg"
        />
      </div>
    </div>
  </div>
</header>
`;
